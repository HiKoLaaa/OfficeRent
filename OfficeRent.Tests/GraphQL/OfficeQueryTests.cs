using System.Threading.Tasks;
using HotChocolate.Execution;
using NUnit.Framework;
using OfficeRent.Tests.Database.TestBuilders.Offices;
using OfficeRent.Tests.GraphQL;
using Snapshooter.NUnit;

namespace OfficeRent.Tests
{
	[TestFixture]
	public class OfficeQueryTests : GraphQLTestsBase
	{
		[Test]
		public async Task GetOffices_RequestThrowGraphQL_AllOfficesGot()
		{
			var office1 = new OfficeTestBuilder()
				.WithAddress(new AddressTestBuilder().WithCity("Moscow").WithStreet("Koroleva").WithStreetNumber("1").Build())
				.WithName("office number 1")
				.WithFloor(1)
				.Build();

			var office2 = new OfficeTestBuilder()
				.WithAddress(new AddressTestBuilder().WithCity("Abakan").WithStreet("Krilova").WithStreetNumber("2").Build())
				.WithName("office number 2")
				.WithFloor(7)
				.Build();

			await AddEntitiesToDbAsync(office1, office2);

			var executor = await ServiceProvider.GetRequestExecutorAsync();

			var request = QueryRequestBuilder
				.New()
				.SetQuery(OfficeQueries.GetAllOffices())
				.SetServices(ServiceProvider)
				.Create();

			var result = await executor.ExecuteAsync(request);

			result.MatchSnapshot();
		}

		[Test]
		public async Task GetOffice_RequestThrowGraphQL_OfficeGot()
		{
			var office1 = new OfficeTestBuilder()
				.WithAddress(new AddressTestBuilder().WithCity("Moscow").WithStreet("Koroleva").WithStreetNumber("1").Build())
				.WithName("office number 1")
				.WithFloor(1)
				.Build();

			var office2 = new OfficeTestBuilder()
				.WithAddress(new AddressTestBuilder().WithCity("Abakan").WithStreet("Krilova").WithStreetNumber("2").Build())
				.WithName("office number 2")
				.WithFloor(7)
				.Build();

			await AddEntitiesToDbAsync(office1, office2);

			var executor = await ServiceProvider.GetRequestExecutorAsync();

			var request = QueryRequestBuilder
				.New()
				.SetQuery(OfficeQueries.GetOffice(office1.Id))
				.SetServices(ServiceProvider)
				.Create();

			var result = await executor.ExecuteAsync(request);

			result.MatchSnapshot();
		}

		private static class OfficeQueries
		{
			public static string GetAllOffices()
			{
				return 
					@"query
{
	offices
	{
		id,
		name,
		address 
		{
			city,
			street,
			streetNumber
		},
		floor
		}
}";
			}

			public static string GetOffice<TId>(TId id)
			{
				return 
					$@"query
{{
	office(id: {id})
	{{
		id,
		name,
		address 
		{{
			city,
			street,
			streetNumber
		}},
		floor
		}}
}}";
			}
		}
	}
}