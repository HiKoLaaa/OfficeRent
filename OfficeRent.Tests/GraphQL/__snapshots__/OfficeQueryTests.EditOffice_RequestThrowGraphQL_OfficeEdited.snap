{
  "Label": null,
  "Path": null,
  "Data": {
    "editOffice": {
      "office": {
        "id": 1,
        "name": "office",
        "floor": 17,
        "address": {
          "street": "Some street 17",
          "city": "city 17",
          "streetNumber": "Beautiful street 17"
        }
      }
    }
  },
  "Errors": null,
  "Extensions": null,
  "ContextData": null,
  "HasNext": null
}
